### PR TITLE
SDIT-1563 Check for a missing booking before validating access

### PR DESCRIPTION
**Pull request by karenmillermoj**
The MissingBookingAccess customEvent was added to temporarily identify access to bookings that don't exist.  Whilst no calls have occurred to trigger this event, we still need to correctly return a 404  if the booking does not exist. This is particularly significant due to the introduction of 403 (rather than 404) when a user does not have access - i.e. we now  separate the 403 and 404 return codes.

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java [hunk 1]
```diff
@@ -565,18 +565,14 @@ private void validateScheduledEventsRequest(final LocalDate fromDate, final Loca
      */
     public void verifyBookingAccess(final Long bookingId, boolean accessDeniedError, final String... rolesAllowed) {
 
+        checkBookingExists(bookingId);
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java:568]: Bomb out straight away if no booking exists - this also does the check for null bookingId

File: src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java [hunk 2]
```diff
@@ -565,18 +565,14 @@ private void validateScheduledEventsRequest(final LocalDate fromDate, final Loca
...
         if (hasAnySystemOverrideRole(rolesAllowed)) return;
 
-        Objects.requireNonNull(bookingId, "bookingId is a required parameter");
-
         final var agencyIds = agencyService.getAgencyIds(false);
         if (AuthenticationFacade.hasRoles("INACTIVE_BOOKINGS")) {
             agencyIds.addAll(Set.of("OUT", "TRN"));
         }
-        // Temporary Logging
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java:576]: no longer needed

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceIntTest_getImageData.kt [hunk 3]
```diff
@@ -4,15 +4,13 @@ package uk.gov.justice.hmpps.prison.api.resource.impl
...
 import org.junit.jupiter.api.Test
-import org.springframework.http.MediaType
 
 @DisplayName("GET /api/bookings/offenderNo/{offenderNo}/image/data")
 class BookingResourceIntTest_getImageData : ResourceTest() {
 
   @Test
   fun `should return 401 when user does not even have token`() {
     webTestClient.get().uri("/api/bookings/offenderNo/A1234AA/image/data")
-      .header("Content-Type", MediaType.APPLICATION_JSON_VALUE)
```
Comment (karenmillermoj) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceIntTest_getImageData.kt:15]: tidy up

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/LocationResourceTest.kt [hunk 4]
```diff
@@ -168,15 +167,13 @@ class LocationResourceTest : ResourceTest() {
       webTestClient.get()
         .uri("/api/locations/description/LEI/inmates?convictedStatus=Convicted")
         .headers(setAuthorisation("ITAG_USER", listOf("VIEW_PRISONER_DATA")))
-        .accept(MediaType.APPLICATION_JSON)
```
Comment (karenmillermoj) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/LocationResourceTest.kt:171]: tidy up

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceImplIntTest_bookingDamageObligations.kt [hunk 5]
```diff
@@ -82,6 +83,7 @@ class OffenderResourceImplIntTest_bookingDamageObligations : ResourceTest() {
   }
 
   private fun stubVerifyOffenderAccess(offenderNo: String) {
+    whenever(bookingRepository.checkBookingExists(anyLong())).thenReturn(true)
```
Comment (karenmillermoj) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceImplIntTest_bookingDamageObligations.kt:86]: Now needed for new check in verifyBookingAccess

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceImplIntTest_getAdjudications.kt [hunk 6]
```diff
@@ -114,7 +112,7 @@ class OffenderResourceImplIntTest_getAdjudications : ResourceTest() {
       val response = testRestTemplate.exchange(
         "/api/offenders/A1234AA/adjudications",
         HttpMethod.GET,
-        createHttpEntityWithBearerAuthorisation("ITAG_USER_ADM", listOf("ROLE_VIEW_ADJUDICATIONS"), Map.of()),
+        createHttpEntityWithBearerAuthorisation("ITAG_USER_ADM", listOf("ROLE_VIEW_ADJUDICATIONS"), mapOf()),
```
Comment (karenmillermoj) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceImplIntTest_getAdjudications.kt:115]: tidy up

File: src/test/java/uk/gov/justice/hmpps/prison/repository/OffenderBookingRepositoryTest.kt [hunk 7]
```diff
@@ -108,7 +107,7 @@ class OffenderBookingRepositoryTest {
       .builder()
       .bookingSequence(1)
       .active(true)
-      .bookingIds(List.of(-1L, -2L))
+      .bookingIds(listOf(-1L, -2L))
```
Comment (karenmillermoj) [inline src/test/java/uk/gov/justice/hmpps/prison/repository/OffenderBookingRepositoryTest.kt:110]: tidy up

File: src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java [hunk 8]
```diff
@@ -554,18 +554,14 @@ private void validateScheduledEventsRequest(final LocalDate fromDate, final Loca
      */
     public void verifyBookingAccess(final Long bookingId, boolean accessDeniedError, final String... rolesAllowed) {
 
+        checkBookingExists(bookingId);
         if (hasAnySystemOverrideRole(rolesAllowed)) return;
 
-        Objects.requireNonNull(bookingId, "bookingId is a required parameter");
```
Comment (karenmillermoj) [inline src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java:559]: Not required as is done as part of the call to checkBookingExists

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**